nomad/fsm.go

```diff
@@ -1931,8 +1888,12 @@ func (s *nomadSnapshot) persistIndexes(sink raft.SnapshotSink,
 		idx := raw.(*state.IndexEntry)
 
 		// Write out a node registration
-		sink.Write([]byte{byte(IndexSnapshot)})
-		if err := encoder.Encode(idx); err != nil {
+		_, err := sink.Write([]byte{byte(IndexSnapshot)})
+		if err != nil {
+			return err
+		}
+		err = encoder.Encode(idx)
+		if err != nil {
```

Comment:
Just fyi - it's perfectly idiomatic in Nomad's source code to use the more compact `if err := func(); err != nil {` approach. You don't need to change it back. Either way is fine.

Reply:
I actually chose to multiline this because I felt the above nil check looked _really_ ugly in a single line, and the inconsistency also made my eye twitch a bit.